nbextensions/appCell2/widgets/appCellWidget.js

```diff
@@ -1963,6 +1800,91 @@ define(
                 return null;
             }
 
+            // LIFECYCLE API
+
+            function init() {
+                return Promise.try(() => {
+                    initializeFSM();
+                    initCodeInputArea();
+
+                    if (!Jupyter.notebook.writable || Jupyter.narrative.readonly) {
+                        setReadOnly();
+                    }
+
+                    return null;
+                });
+            }
+
+            function attach(node) {
+                return Promise.try(() => {
+                    hostNode = node;
+                    container = hostNode.appendChild(document.createElement('div'));
+                    ui = UI.make({
+                        node: container,
+                        bus,
+                    });
+
+                    actionButtonWidget = ActionButtons.make({
+                        ui,
+                        actionButtons,
+                        bus,
+                        runAction: doActionButton,
+                        cssCellType: null,
+                    });
+
+                    const layout = renderLayout();
+                    container.innerHTML = layout.content;
+                    layout.events.attachEvents(container);
+                    $(container).find('[data-toggle="popover"]').popover();
+                    return null;
+                }).catch((error) => {
+                    throw new Error('Unable to attach app cell widget: ' + error);
+                });
+            }
+
+            function start() {
```

Comment:
copied as-is apart from the `addBusEvents()` method subbed in for the long string of bus events added inline